### ISSUE TITLE
viafree.se svtplay-dl crashes when trying to download videos

### ISSUE THREAD

**Issue by qnorsten** [spaam/svtplay-dl#431]
I am running these commands using the latest master. Tried it on both windows and linux. Tried a few different videos two examples are below.
svtplay-dl --remux -s -v http://www.viafree.se/program/livsstil/hells-kitchen/sasong-12/avsnitt-1
svtplay-dl --remux -s -v  http://www.viafree.se/program/livsstil/property-ladder-revisited/sasong-2/avsnitt-6

I get the following error. 

> DEBUG [1471872644.180573] /usr/local/lib/python3.4/dist-packages/svtplay_dl-1.2-py3.4.egg/svtplay_dl/utils/__init__.py/request: HTTP getting 'http://www.viafree.se/program/livsstil/property-ladder/sasong-3/avsnitt-7'
> ERROR [1471872644.271976] /usr/local/lib/python3.4/dist-packages/svtplay_dl-1.2-py3.4.egg/svtplay_dl/__init__.py/get_one_media: version: 1.2
> Traceback (most recent call last):
>   File "/usr/local/bin/svtplay-dl", line 5, in <module>
>     pkg_resources.run_script('svtplay-dl==1.2', 'svtplay-dl')
>   File "/usr/lib/python3/dist-packages/pkg_resources.py", line 528, in run_script
>     self.require(requires)[0].run_script(script_name, ns)
>   File "/usr/lib/python3/dist-packages/pkg_resources.py", line 1394, in run_script
>     execfile(script_filename, namespace, namespace)
>   File "/usr/lib/python3/dist-packages/pkg_resources.py", line 55, in execfile
>     exec(compile(open(fn).read(), fn, 'exec'), globs, locs)
>   File "/usr/local/lib/python3.4/dist-packages/svtplay_dl-1.2-py3.4.egg/EGG-INFO/scripts/svtplay-dl", line 7, in <module>
>     svtplay_dl.main()
>   File "/usr/local/lib/python3.4/dist-packages/svtplay_dl-1.2-py3.4.egg/svtplay_dl/__init__.py", line 438, in main
>     get_media(url, options)
>   File "/usr/local/lib/python3.4/dist-packages/svtplay_dl-1.2-py3.4.egg/svtplay_dl/__init__.py", line 171, in get_media
>     get_one_media(stream, options)
>   File "/usr/local/lib/python3.4/dist-packages/svtplay_dl-1.2-py3.4.egg/svtplay_dl/__init__.py", line 218, in get_one_media
>     for i in streams:
>   File "/usr/local/lib/python3.4/dist-packages/svtplay_dl-1.2-py3.4.egg/svtplay_dl/service/viaplay.py", line 89, in get
>     vid = self._get_video_id()
>   File "/usr/local/lib/python3.4/dist-packages/svtplay_dl-1.2-py3.4.egg/svtplay_dl/service/viaplay.py", line 73, in _get_video_id
>     if len(n["episodeNumber"]) > 0 and int(episodenr) == n["episodeNumber"]:
> TypeError: object of type 'int' has no len()

**Comment by agent666** [comment]
+1

Sedan senaste update så funkar inte "serie" episoder igen..

**Comment by larsamannen** [comment]
Same problem.
Downloading from viafree.se works with following commit:
e0caf5c8228717740e0f00f34f9d42246759fa29

**Comment by Sopor-** [comment]
On latest master i get a crash when trying to use -S and -A, but it seems to crash without -S and A too:
```
$ svtplay-dl -S -A http://www.viafree.se/program/drama/finding-carter --verbose
DEBUG [1471985532.19] /usr/local/lib/python2.7/dist-packages/svtplay_dl-1.2-py2.7.egg/svtplay_dl/utils/__init__.py/request: HTTP getting 'http://www.viafree.se/program/drama/finding-carter'
INFO [1471985532.22] /usr/local/lib/python2.7/dist-packages/svtplay_dl-1.2-py2.7.egg/svtplay_dl/__init__.py/get_all_episodes: Episode 1 of 11
DEBUG [1471985532.22] /usr/local/lib/python2.7/dist-packages/svtplay_dl-1.2-py2.7.egg/svtplay_dl/utils/__init__.py/request: HTTP getting u'http://www.tv3play.se/program/finding-carter/743172'
ERROR [1471985532.34] /usr/local/lib/python2.7/dist-packages/svtplay_dl-1.2-py2.7.egg/svtplay_dl/__init__.py/get_one_media: version: 1.2
Traceback (most recent call last):
  File "/usr/local/bin/svtplay-dl", line 4, in <module>
    __import__('pkg_resources').run_script('svtplay-dl==1.2', 'svtplay-dl')
  File "/usr/local/lib/python2.7/dist-packages/pkg_resources/__init__.py", line 719, in run_script
    self.require(requires)[0].run_script(script_name, ns)
  File "/usr/local/lib/python2.7/dist-packages/pkg_resources/__init__.py", line 1504, in run_script
    exec(code, namespace, namespace)
  File "/usr/local/lib/python2.7/dist-packages/svtplay_dl-1.2-py2.7.egg/EGG-INFO/scripts/svtplay-dl", line 7, in <module>
    svtplay_dl.main()
  File "/usr/local/lib/python2.7/dist-packages/svtplay_dl-1.2-py2.7.egg/svtplay_dl/__init__.py", line 441, in main
    get_media(url, options)
  File "/usr/local/lib/python2.7/dist-packages/svtplay_dl-1.2-py2.7.egg/svtplay_dl/__init__.py", line 172, in get_media
    get_all_episodes(stream, options, url)
  File "/usr/local/lib/python2.7/dist-packages/svtplay_dl-1.2-py2.7.egg/svtplay_dl/__init__.py", line 200, in get_all_episodes
    get_one_media(substream, copy.copy(options))
  File "/usr/local/lib/python2.7/dist-packages/svtplay_dl-1.2-py2.7.egg/svtplay_dl/__init__.py", line 221, in get_one_media
    for i in streams:
  File "/usr/local/lib/python2.7/dist-packages/svtplay_dl-1.2-py2.7.egg/svtplay_dl/service/viaplay.py", line 89, in get
    vid = self._get_video_id()
  File "/usr/local/lib/python2.7/dist-packages/svtplay_dl-1.2-py2.7.egg/svtplay_dl/service/viaplay.py", line 73, in _get_video_id
    if len(n["episodeNumber"]) > 0 and int(episodenr) == n["episodeNumber"]:
TypeError: object of type 'int' has no len()
```

**Comment by magic75** [comment]
git bisect säger...

c799375e735aaac1aa2590ebe29454e35b66ac4a is the first bad commit
commit c799375e735aaac1aa2590ebe29454e35b66ac4a
Author: Johan Andersson <j@i19.se>
Date:   Sun Aug 21 01:08:39 2016 +0200

    viaplay: more improvements for viafree with single videos
    
    fixes #427
